lib.php

```diff
@@ -264,3 +266,100 @@ function add_event_to_sent_log($event) {
         $DB->insert_record('logstore_xapi_sent_log', $newrow);
     }
 }
+
+/**
+ * Extract events from logstore_xapi_log or logstore_xapi_failed_log.
+ *
+ * @return array
+ */
+function logstore_xapi_extract_events($limitnum = 0, $log = XAPI_REPORT_SOURCE_LOG, $type = null) {
+    global $DB;
+
+    $conditions = null;
+    if ($type) {
+        $conditions = array("type" => $type);
+    }
+    $sort = '';
+    $fields = '*';
+    $limitfrom = 0;
+
+    if ($log == XAPI_REPORT_SOURCE_LOG) {
+        $events = $DB->get_records('logstore_xapi_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
+    } else {
+        $events = $DB->get_records('logstore_xapi_failed_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
+    }
```

Comment:
In an earlier round of comments I suggested changes
https://github.com/xAPI-vle/moodle-logstore_xapi/pull/726#discussion_r434141978

They have all been done except the last part to use the `$log` variable in the call to `$DB->get_records`.

Reply:
I think the name of the table should be in this data access method and not passed to the method.

Reply:
I have changed this to use the param as the table name, but a good reason for this not to be a table name is that if it was called in a tight loop it would cause an unnecessary string comparison every time the function is called. It's not so bad if the input parameter is an enumeration as an int compare is hundreds of times faster than a string compare.